kong/llm/drivers/gemini.lua

```diff
@@ -292,7 +292,24 @@ function _M.subrequest(body, conf, http_opts, return_res_table)
     ["Content-Type"] = "application/json",
   }
 
-  if conf.auth and conf.auth.header_name then
+  if identity_interface and identity_interface.interface then
+    if identity_interface.interface:needsRefresh() then
+      -- HACK: A bug in lua-resty-gcp tries to re-load the environment
+      --       variable every time, which fails in nginx
+      --       Create a whole new interface instead.
+      --       Memory leaks are mega unlikely because this should only
+      --       happen about once an hour, and the old one will be
+      --       cleaned up anyway.
+      local service_account_json = identity_interface.interface.service_account_json
+      local identity_interface_new = identity_interface.interface:new(service_account_json)
```

Comment:
repeatedly creating variables like above may lead to garbage collector pressure in some cases (eg, if that happens a lot and if the scope is long-lived). if the new variable is only referenced once, like here, it's better to just reference the existing variable